ecosystem/typescript/sdk/src/tests/e2e/indexer.test.ts

```diff
@@ -191,5 +193,20 @@ describe("Indexer", () => {
       const ledgerInfo = await indexerClient.getIndexerLedgerInfo();
       expect(ledgerInfo.ledger_infos[0].chain_id).toBeGreaterThan(1);
     });
+
+    test("make sure queryIndexer post call with correct User-Agent header", async () => {
+      let mockHeaders: any = {};
+
+      // spy on axios.post and mock the response
+      jest.spyOn(axios, 'post').mockImplementation((url, query, config) => {
```

Comment:
can we test it for when it does a real call instead of mocking it?

Reply:
Yep, but I wonder though, if we should do the whole integration test or just checking the header would be sufficient?

Reply:
I don't think we have to do active testing.  Please test it once with a real world situation and show it, then mocking should be fine, as long as the headers show the same.

Reply:
> I don't think we have to do active testing. Please test it once with a real world situation and show it, then mocking should be fine, as long as the headers show the same.

Wanted to make sure I understand this correctly. Basically stay with mocking,  but also manually test it with real call and and show/record the result?

Reply:
Update - Manually checked and make sure that `User-Agent` is included from the request.